src/incomingDelivery/controller.js

```diff
@@ -517,6 +519,48 @@ function getAllReceived(req, res) {
     "getting all received incoming deliveries"
   );
 }
+
+function setCanceled(req, res) {
+  ExpressHandler(
+    async () => {
+      const { _id, reason } = req.body;
+
+      if (!reason) return HTTPError(`Reason is required.`, 400);
+
+      const incomingDelivery = await IncomingDelivery.findById(_id);
```

Comment:
I think if you put the middleware up above, alot of these checks can go away

Reply:
Done.